### DISCUSSION TITLE
Option to clamp number input value on change instead of only blur

### DISCUSSION THREAD

**Discussion by juliomuhlbauer** [chakra-ui/zag#1097] [Ideas]
When typing number like 120000,  I want the user to see the formatted number when typing the number instead of only on blur.
Seeing like 120,000 is better than 120000.

I have done it manually, but it would be great implemented out of the box in zag

![image](https://github.com/chakra-ui/zag/assets/53458125/3b8d7dd9-e7cd-4642-9bde-2a4ab4caeb23)

**Comment by segunadebayo** [comment]
Closing as outdated — please refer to the updated v1.0 number-input API. If this is still relevant, please open a new discussion.
